redwood-lazylayout-compose/src/commonTest/kotlin/app/cash/redwood/lazylayout/compose/LazyListTest.kt

```diff
@@ -86,6 +98,136 @@ class LazyListTest {
         ),
       )
   }
+
+  @OptIn(RedwoodCodegenApi::class)
+  @Test
+  fun scrollPopulatedLazyColumn() = runTest {
+    val windowedLazyList = FakeWindowedLazyList()
+    val factories = TestSchemaWidgetFactories(
+      TestSchema = TestSchemaTestingWidgetFactory(),
+      RedwoodLayout = RedwoodLayoutTestingWidgetFactory(),
+      RedwoodLazyLayout = RedwoodLazyLayoutFakeWidgetFactory(windowedLazyList),
+    )
+    val snapshot = TestSchemaTester(factories) {
+      setContent {
+        LazyColumn(placeholder = { Text("Placeholder") }) {
+          items(100) {
+            Text(it.toString())
+          }
+        }
+      }
+      awaitSnapshot()
+      windowedLazyList.updateViewport(50, 60)
+      awaitSnapshot()
+    }
+
+    assertThat(snapshot)
+      .containsExactly(
+        DefaultLazyListValue.copy(
+          itemsBefore = 35,
+          itemsAfter = 25,
+          placeholder = List(20) { TextValue(Modifier, "Placeholder") },
+          items = List(40) { TextValue(Modifier, (it + 35).toString()) },
+        ),
+      )
+  }
+}
+
+@OptIn(RedwoodCodegenApi::class)
+private suspend fun <R> TestSchemaTester(
+  factories: TestSchemaWidgetFactories<WidgetValue> = TestSchemaWidgetFactories(
+    TestSchema = TestSchemaTestingWidgetFactory(),
+    RedwoodLayout = RedwoodLayoutTestingWidgetFactory(),
+    RedwoodLazyLayout = RedwoodLazyLayoutTestingWidgetFactory(),
+  ),
+  body: suspend TestRedwoodComposition<List<WidgetValue>>.() -> R,
+): R =
```

Comment:
Isn't this supposed to be generated for the testing code automatically?

Reply:
I commented on this but I forgot to press submit 🤦  This is what I originally wrote about it:

> This is a copy-paste of the generated `TestSchemaTester`, with the only change being that factories is now a parameter, thus can be changed. Would such an API change make sense within the generated code?

For reference, this is what the generated one looks like:

```
@OptIn(RedwoodCodegenApi::class)
public suspend fun <R>
    TestSchemaTester(body: suspend TestRedwoodComposition<List<WidgetValue>>.() -> R): R =
    coroutineScope {
  val factories = TestSchemaWidgetFactories(
    TestSchema = TestSchemaTestingWidgetFactory(),
    RedwoodLayout = RedwoodLayoutTestingWidgetFactory(),
    RedwoodLazyLayout = RedwoodLazyLayoutTestingWidgetFactory(),
  )
  val container = MutableListChildren<WidgetValue>()
  val tester = TestRedwoodComposition(this, factories, container,
      MutableStateFlow(UiConfiguration())) {
    container.map { it.value }
  }
  try {
    tester.body()
  } finally {
    tester.cancel()
  }
}
```

Reply:
Yeah I'll think about this tomorrow

Reply:
I'm gonna merge this for now. I'm happy to create a follow-up PR that updates the generation of `TestSchemaTester` such that this custom `TestSchemaTester` can be deleted, either by updating it as it is here, or in a different way that'd allow custom factory implementations to be passed in.